mobile/examples/phi-3/android/app/src/main/java/ai/onnxruntime/genai/demo/MainActivity.java

```diff
@@ -85,8 +108,29 @@ public void onClick(View v) {
                     @Override
                     public void run() {
                         try {
-                            genAIWrapper.run(promptQuestion_formatted);
-                        } catch (GenAIException e) {
+                            TokenizerStream stream = tokenizer.createStream();
+
+                            GeneratorParams generatorParams = model.createGeneratorParams();
+                            //generatorParams.setSearchOption("length_penalty", 1000);
+                            generatorParams.setSearchOption("max_length", 500);
+
+                            Sequences encodedPrompt = tokenizer.encode(promptQuestion_formatted);
+                            generatorParams.setInput(encodedPrompt);
+
+                            Generator generator = new Generator(model, generatorParams);
+
+                            while (!generator.isDone()) {
+                                generator.computeLogits();
+                                generator.generateNextToken();
+                 
+                                int token = generator.getLastTokenInSequence(0);
+                 
+                                tokenListener.accept(stream.decode(token));
+                            }
+
+                            //generator.close();
+                        }
+                        catch (GenAIException e) {
                             throw new RuntimeException(e);
```

Comment:
having an uncaught exception from a thread may not behave well. maybe we should at least log it. even better if the error can be shown to the user.